Assets/Scripts/Entities/Enemy.cs

```diff
@@ -127,6 +129,12 @@ public void Die(bool isDespawning = false)
                 SpawnLoot();
             }
 
+            if (EcRef == default)
+            {
+                EcRef = FindObjectOfType<EnemyController>();
+            }
+
+            EcRef.livingEnemies.Remove(this);
```

Comment:
This needs to be tested - the livingEnemies list may need to be flushed or this val might need to be nullified to work properly.